tree/dataframe/inc/ROOT/RDF/RInterface.hxx

```diff
@@ -2410,13 +2410,15 @@ private:
 
       auto snapHelperArgs = std::make_shared<RDFInternal::SnapshotHelperArgs>(RDFInternal::SnapshotHelperArgs{
          std::string(filename), std::string(dirname), std::string(treename), columnList, options});
-      auto actionPtr =
-         RDFInternal::BuildAction<ColumnTypes...>(validCols, snapHelperArgs, fLoopManager->GetNSlots(), fProxiedPtr,
-                                                  RDFInternal::ActionTags::Snapshot{}, fDefines);
-      fLoopManager->Book(actionPtr.get());
 
-      return RDFInternal::CreateSnapshotRDF(validCols, fullTreename, filename, options.fLazy, *fLoopManager,
-                                            std::move(actionPtr));
+      ::TDirectory::TContext ctxt;
```

Comment:
Has the `TContext` some magic behind the ctor or is this variable forgotten?

Reply:
Magic :) https://root.cern/doc/master/classTDirectory_1_1TContext.html

Reply:
I think you miss the include for `TDirectory.h`?

Reply:
fixed